app/channel.go

```diff
@@ -382,47 +407,51 @@ func (a *App) createDirectChannel(userID string, otherUserId string) (*model.Cha
 
 	result = <-uc2
 	if result.NErr != nil {
-		return nil, model.NewAppError("CreateDirectChannel", "api.channel.create_direct_channel.invalid_user.app_error", nil, otherUserId, http.StatusBadRequest)
+		return nil, model.NewAppError("CreateDirectChannel", "api.channel.create_direct_channel.invalid_user.app_error", nil, otherUserID, http.StatusBadRequest)
 	}
 	otherUser := result.Data.(*model.User)
 
-	channel, nErr := a.Srv().Store.Channel().CreateDirectChannel(user, otherUser)
-	if nErr != nil {
+	return a.createDirectChannelWithUser(user, otherUser)
+}
+
+func (a *App) createDirectChannelWithUser(user, otherUser *model.User) (*model.Channel, *model.AppError) {
+	channel, err := a.Srv().Store.Channel().CreateDirectChannel(user, otherUser)
+	if err != nil {
 		var invErr *store.ErrInvalidInput
 		var cErr *store.ErrConflict
 		var ltErr *store.ErrLimitExceeded
 		var appErr *model.AppError
 		switch {
-		case errors.As(nErr, &invErr):
+		case errors.As(err, &invErr):
 			switch {
 			case invErr.Entity == "Channel" && invErr.Field == "DeleteAt":
-				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save.archived_channel.app_error", nil, "", http.StatusBadRequest)
+				return nil, model.NewAppError("CreateDirectChannel", "store.sql_channel.save.archived_channel.app_error", nil, "", http.StatusBadRequest)
```

Comment:
Should all these be `model.NewAppError("createDirectChannelWithUser"` instead?

Reply:
Yeah. Let me change them.